Modules/Forum/classes/class.ilForumCronNotification.php

```diff
@@ -230,6 +232,16 @@ public function sendCronForumNotification(ilDBStatement $res, int $notification_
             }
 
             $row['ref_id'] = $ref_id;
+            $top_item_crs_ref = $this->tree->checkForParentType($ref_id, 'crs');
```

Comment:
We should not execute both operations here. We should check for a `crs` parent first. If a `crs` could **not** be found, we should check for a `grp`.

IMO we still need caching here. I suggest to extract this into a new method, nesting `if` statements should be avoided. We should work with early returns here.